src/systemctl/systemctl-set-default.c

```diff
@@ -51,7 +51,8 @@ static int determine_default(char **ret_name) {
         if (install_client_side()) {
                 r = unit_file_get_default(arg_scope, arg_root, ret_name);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to get default target: %m");
+                        return log_error_errno(r, "Failed to get default target: %s",
+                                               r == -ERFKILL ? "Unit is masked" : strerror_safe(r));
```

Comment:
plese don't use strerror()/strerror_safe() when you might as well use %m.

The thing is that strerror() is not thread-safe, it uses a static process-wider buffer. While systemctl is not threaded, we still try to avoid thread-unsafe APIs because you never know what kind of bg stuff libraries do, and we want our stuff to be something we can move around in the tree with least amount of headaches. LTDR, plese rework as:

```c
if (r == -ERFKILL)
        return log_error_errno(r, "Failed to get default target, as it is masked.");
if (r < 0)
        return log_error_errno(r, "Failed to get default target: %m");
```

you get the idea.

Reply:
I thought this might be preferred, but tried the short version first... Updated.